subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidatePlugins.java

```diff
@@ -76,120 +54,56 @@
  */
 @CacheableTask
 @Incubating
-public class ValidatePlugins extends DefaultTask {
+public abstract class ValidatePlugins extends DefaultTask {
     private final ConfigurableFileCollection classes;
     private final ConfigurableFileCollection classpath;
     private final RegularFileProperty outputFile;
     private final Property<Boolean> enableStricterValidation;
     private final Property<Boolean> ignoreFailures;
     private final Property<Boolean> failOnWarning;
 
-    @Inject
-    public ValidatePlugins(ObjectFactory objects) {
-        this.classes = objects.fileCollection();
-        this.classpath = objects.fileCollection();
-        this.outputFile = objects.fileProperty();
-        this.enableStricterValidation = objects.property(Boolean.class).convention(false);
-        this.ignoreFailures = objects.property(Boolean.class).convention(false);
-        this.failOnWarning = objects.property(Boolean.class).convention(true);
+    public ValidatePlugins() {
+        this.classes = getObjects().fileCollection();
+        this.classpath = getObjects().fileCollection();
+        this.outputFile = getObjects().fileProperty();
+        this.enableStricterValidation = getObjects().property(Boolean.class).convention(false);
+        this.ignoreFailures = getObjects().property(Boolean.class).convention(false);
+        this.failOnWarning = getObjects().property(Boolean.class).convention(true);
     }
 
     @TaskAction
     public void validateTaskClasses() throws IOException {
-        ClassLoader previousContextClassLoader = Thread.currentThread().getContextClassLoader();
-        ClassPath classPath = DefaultClassPath.of(Iterables.concat(getClasses(), getClasspath()));
-        ClassLoader classLoader = getClassLoaderFactory().createIsolatedClassLoader("task-loader", classPath);
-        Thread.currentThread().setContextClassLoader(classLoader);
-        try {
-            validateTaskClasses(classLoader);
-        } finally {
-            Thread.currentThread().setContextClassLoader(previousContextClassLoader);
-            ClassLoaderUtils.tryClose(classLoader);
-        }
-    }
+        getWorkerExecutor()
+            .classLoaderIsolation(spec -> spec.getClasspath().setFrom(getClasses(), getClasspath()))
+            .submit(ValidateAction.class, params -> {
+                params.getClasses().setFrom(getClasses());
+                params.getOutputFile().value(getOutputFile());
```

Comment:
Any particular reason why you are using `value()` and not `set()` for setting the property?

Reply:
Changed to `set()`.